### PR TITLE
[Limit orders] confirm screen

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

image

## Test
1. Select valid trade (input/output currencies and amounts)
2. Ensure that `/api/v1/quote` didn't respose with 400 error (we will get rid of the call in the future)
3. Click `Trade`

### CODE REVIEW

File: src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx
Comment (alfetopito) [inline src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx:40]: Hmm?

File: src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx:30]: Can you move this calculation outside of the component?
Actually, still using `Number` which will be replaced by whatever BigNumber we pick, so, might be ok to be here for now

File: src/cow-react/modules/limitOrders/pure/LimitOrdersConfirm/index.tsx
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/pure/LimitOrdersConfirm/index.tsx:30]: Shouldn't this change according to order type?

**Comment by alfetopito** [comment]
How to test? Should it be visible in the UI?

File: src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx
Comment (shoom3301) [reply src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx:40]: We have `readOnly={true}` above, however, the `NumericalInput` component interface require `onUserInput` prop and i just added empty function

**Comment by shoom3301** [comment]
@alfetopito added test instruction

**Comment by elena-zh** [comment]
Hey @shoom3301 , nice changes!
Some issues I faced: 
1. there is now arrow between from and to fields on the modal
image

2. There is no 'recipient' info when send an order to a recipient
Screenshot_5
image
3. The date format looks a bit weird (in brackets, the time is moving)
why date in

4. The button should be 'Submit limit order' instead of the 'Confirm'
5. The modal is not closed when cancel/sigh a transaction in the connected wallet
6. I think, it would be great to show full amounts on hover like we do on regular swap modals
image

Thanks!

**Comment by shoom3301** [comment]
@elena-zh thank you!
2 and 5 fixed, the reset will be fixed after applying design (in the future)

**Comment by elena-zh** [comment]
Hey @shoom3301 , I don't see changes for cases 2,5. 
Seems that the build failed.

**Comment by elena-zh** [comment]
I have also noticed that the modal may appear after filling in an amount. See the video:  https://www.loom.com/share/08d79903f68147e7ac81f9183b5f9a3e

But I believe that it is going to be fixed when we implement Trade button validation (I mean keep the button disabled when a token pair is not selected/amount is not filled/price is not filled/etc.)

**Comment by elena-zh** [comment]
Hey @shoom3301 , case 2 looks awesome!
As for the case 5, I have a nitpick: it would be great to show a full amount on a price hover as well
image

Thanks!

File: src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx
Comment (alfetopito) [inline src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx:37]: There's a util right that was created to deal with Fraction instances.
Not sure it'll fit the type, though https://github.com/cowprotocol/cowswap/blob/6f5e9a33f87ebb93fbbeb73dcbdd60e0f06376e0/src/custom/utils/format.ts#L171

File: src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx
Comment (alfetopito) [inline src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx:25]: We definitely need to think about this as I'm not comfortable with this exponential operation using JS numbers 😰

**Review by alfetopito — APPROVED** [review]
Code wise ok with minor comments

On behaviour, waiting for Elena's seal of approval

File: src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx
Comment (shoom3301) [reply src/cow-react/common/pure/CurrencyInputPanel/CurrencyPreview.tsx:37]: Here we should display an exact amount, so I think it's correct

File: src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/containers/LimitOrdersConfirmModal/index.tsx:25]: We will cooperate with @nenadV91 and refactor rate-related values to BigNumber

**Comment by shoom3301** [comment]
@elena-zh Thanks!
Added title here: https://github.com/cowprotocol/cowswap/pull/1238

**Comment by elena-zh** [comment]
> @elena-zh Thanks! Added title here: #1238

It shows an incorrect amount on hover now. Reported #1318 issue for this

**Pull request merged**
